src/scaffoldfitter/fitterstepalign.py

```diff
@@ -275,24 +276,76 @@ def _optimiseAlignment(self, pointMap):
             nodetemplate.defineField(modelCoordinates)
             nodetemplate.defineField(dataCoordinates)
             fieldcache = fieldmodule.createFieldcache()
+
+            modelsum = [0.0, 0.0, 0.0]
+            datasum = [0.0, 0.0, 0.0]
+            modelMin = copy.deepcopy(list(pointMap.values())[0][0])
+            modelMax = copy.deepcopy(list(pointMap.values())[0][0])
+            dataMin =  copy.deepcopy(list(pointMap.values())[0][1])
+            dataMax = copy.deepcopy(list(pointMap.values())[0][1])
             for name, positions in pointMap.items():
                 modelx = positions[0]
                 datax = positions[1]
+                modelsum = [(modelsum[c] + modelx[c]) for c in range(3)]
+                datasum = [(datasum[c] + datax[c]) for c in range(3)]
                 node = nodes.createNode(-1, nodetemplate)
                 fieldcache.setNode(node)
                 result1 = modelCoordinates.assignReal(fieldcache, positions[0])
                 result2 = dataCoordinates.assignReal(fieldcache, positions[1])
                 assert (result1 == RESULT_OK) and (result2 == RESULT_OK), "Align:  Failed to set up data for alignment to markers optimisation"
+                for c in range(3):
+                    modelMin[c] = min(modelx[c], modelMin[c])
+                    modelMax[c] = max(modelx[c], modelMax[c])
+                    dataMin[c] = min(datax[c], dataMin[c])
+                    dataMax[c] = max(datax[c], dataMax[c])
+
+            groupCount = len(pointMap)
```

Comment:
len(pointMap) will be an integer which will be converted to float in all the following expressions. In performance critical code (C/C++) division is still slower than multiplication so it's usually best to get a scalefactor 1.0/value and multiply by it. In this case, I'd like to see it become a float:
```
groupScale = 1.0/len(pointMap)
modelCM = [(s*groupScale) for s in modelsum]
dataCM = [(s*groupScale) for s in datasum]
```
Here I use the s for a scalar in a list. Then keep variable c for component indexes.